### PR TITLE
Signup: Add transitions to step one survey

### PR DESCRIPTION
**Pull request by shaunandrews**
The first step of signup (called the "survey") asks users to select two options: a vertical and a sub-vertical. The survey UI shows one list of options at a time, replacing the first with the second. The way this works now is a little jarring:

survey-transition-current

This PR adds some a transition between the first vertical options and the second sub-vertical options:

untitled

Test live: https://calypso.live/start/?branch=update/signup-survey

### CODE REVIEW

File: client/signup/steps/survey/style.scss [hunk 1]
```diff
@@ -3,44 +3,76 @@
...
 	color: lighten( $gray, 10% );
-}
 
-.survey-step__vertical:hover .survey-step__vertical__icon {
-	color: $gray;
-}
+	.survey-step__vertical:hover & {
+		color: $gray;
+	}
+}
```
Comment (bisko) [inline client/signup/steps/survey/style.scss]: There should be a new line at the end of the file.

**Comment by folletto** [comment]
Design wise, it's a great idea. 👍 

Tested on Chrome, Firefox, Safari, Edge, IE11. Works well.

Edge and IE11 have no blur, but still it's cool.

**Comment by coreh** [comment]
Minor nitpick: The back animation has a small bug. Since we clear `this.state.stepOne`, the div where the second step is rendered becomes empty during the back animation playback. You can't notice this really well with a 0.2s delay, but here's what happens in slow motion:

https://cloudup.com/i0WBO0zqpZa

**Comment by shaunandrews** [comment]
> The back animation has a small bug. Since we clear this.state.stepOne, the div where the second step is rendered becomes empty during the back animation playback.

Totally. I gave the sub-verticals a min-height to help, since without it the div was collapsing when cleared. Its a minor detail, but if you have any ideas for how to resolve it, my ears are wide open. Maybe we can delay the "clearing" for 0.2s or something to accommodate the transition?

**Comment by rickybanister** [comment]
Fancy!

**Comment by coreh** [comment]
LGTM 👍

**Comment by michaeldcain** [comment]
LGTM. Maybe squash, then :shipit:

**Pull request merged**